### ISSUE TITLE
User should be able to write `:smile:` for 😄 and use other emoji codes

### ISSUE THREAD

**Issue by pdurbin** [TryQuiet/monorepo#540]
As a keyboard junkie, in Element, Slack, and GitHub I very frequently write `:smile:` to get the 😄 emoji. I would prefer not to reach for the mouse to click the emoji button in Quiet:

<img width="184" alt="Screen Shot 2022-05-27 at 9 31 57 AM" src="https://user-images.githubusercontent.com/21006/170709461-4b162a57-a7e2-4162-a4c7-fc2cea461c32.png">

I'm aware that a workaround on Mac is to hit Ctrl-Cmd-Spacebar to bring up a native emoji picker...

<img width="788" alt="Screen Shot 2022-05-27 at 9 35 14 AM" src="https://user-images.githubusercontent.com/21006/170710150-ea0f2fd0-43ff-447c-b244-dd1a27ed4067.png">

... but it's faster and easier for me to just type `:smile:` like I do in other places.

Tab completion and search/autocomplete would be nice (see https://github.blog/2012-10-12-emoji-autocomplete/ ). That is, here on GitHub I only have to type `:smi` and then hit the Tab key to select from a list. Element and Slack work the same way.

![Screen Shot 2022-05-27 at 9 45 52 AM](https://user-images.githubusercontent.com/21006/170712012-0f448ac2-31f1-4912-8bdc-0fe252f86062.png)

Some places such as https://github.com/ikatyang/emoji-cheat-sheet seem to refer to these as emoji shortcodes for anyone searching issues for that term.

**Comment by holmesworcester** [comment]
Thanks for this! 

Just noting for when we get to this that the visual autocomplete interacts with @ mentions, as the UI is similar, so we might want to tackle it at the same time. This library might work for both, or we could use the existing @ mention code, though it might not consider all the cases: https://github.com/webscopeio/react-textarea-autocomplete. 

We could also look at what other editors or open source chat apps use for this.

**Comment by pdurbin** [comment]
https://github.com/nolanlawson/emoji-picker-element seems promising. I heard about it at https://shoptalkshow.com/451/ when they interviewed @nolanlawson.

**Comment by holmesworcester** [comment]
This seems like a good proposal! https://chat.openai.com/share/1491b99e-d50c-463f-af00-d8e389644bc1

**Comment by holmesworcester** [comment]
@pdurbin should we assign this to you?

**Comment by pdurbin** [comment]
@holmesworcester no, I'm not actively working on it. I just closed my draft pull request:

- #1822

I'm also happy to close this issue if it helps keeps things tidy.

By the way, https://messages.google.com suffers from the same problem. Typing `:smile:` doesn't create an emoji:

<img width="465" alt="Screenshot 2023-10-11 at 7 58 44 AM" src="https://github.com/TryQuiet/quiet/assets/21006/414d3cbe-69ee-4626-9621-fa00a811dbf2">

**Comment by agiledev24** [comment]
Implemented this feature and created a PR: https://github.com/TryQuiet/quiet/pull/2354

Followed discord design.
@holmesworcester 
![Image](https://github.com/TryQuiet/quiet/assets/80997421/0c39e54e-1e7a-4d69-8e79-ab40ca6f2eff)

**Comment by holmesworcester** [comment]
Thank you @agiledev24 !!!

**Comment by holmesworcester** [comment]
@leblowl how easy is it to make a build for kinda so that she can test this feature and provide feedback or approve quickly?

@jgaylor do you have any feedback on the design? We could have a look at this Friday too.

**Comment by leblowl** [comment]
@holmesworcester It's easy, but mobile is sort of broken at the moment for @kingalg due to the app crashing, I'm looking into it.

**Comment by holmesworcester** [comment]
What about starting a desktop-only release?

**Comment by leblowl** [comment]
Sure, we can

**Comment by leblowl** [comment]
We generally have been merging into develop before releasing, so I'd probably need to change some things if we want to release this and test before merging. Also we've generally been releasing one thing at a time and 2.1.2 is getting backed up due to some bugs that @kingalg found, so I'll have to think about how we can handle this.

**Comment by kingalg** [comment]
@holmesworcester In our current situation, I would advise waiting to fix mobiles before releasing. Normally, it wouldn't be a huge problem releasing only one feature for part of the platforms, but we bumped React Native in this version, and there was also some refactoring. It would be better to have it all checked together because there may be some unexpected things.

**Comment by holmesworcester** [comment]
Addressed in: https://github.com/TryQuiet/quiet/pull/2776

**Comment by ethanc8** [comment]
It would be nice to have this functionality on mobile too -- although it's probably way less common for most users to use the colon codes for emojis on mobile I still tend to use them on mobile.

**Comment by holmesworcester** [comment]
> It would be nice to have this functionality on mobile too -- although it's probably way less common for most users to use the colon codes for emojis on mobile I still tend to use them on mobile.

@ethanc8 That's fascinating! And you would expect the same functionality as on desktop? I'll break this out as a separate issue.

**Comment by kingalg** [comment]
Version: 4.1.0-alpha.0
System: iOS, Linux and Windows. 

Done.
